OpenRA.Mods.Common/Traits/CashTrickler.cs

```diff
@@ -84,21 +84,10 @@ void AddCashTick(Actor self, int amount)
 
 		void ModifyCash(Actor self, Player newOwner, int amount)
 		{
-			if (amount < 0)
-			{
-				// Check whether the amount of cash to be removed would exceed available player cash, in that case only remove all the player cash
-				var drain = Math.Min(resources.Cash + resources.Resources, -amount);
-				resources.TakeCash(drain);
+			resources.ChangeCash(amount);
```

Comment:
Here and in `GivesCashOnCapture` the amount is first tweaked so that it will take the player to 0 cash (otherwise `TakeCash` won't do anything) and then this tweaked amount is shown in the cash tick.

I'm pretty sure that we want this same behaviour for the other traits that now use `ChangeCash`, so could you please add this to the `else` branch of `ChangeCash` and make that method return the actual changed amount?  The callers can then use this return value in their cash ticks.

Reply:
This needs to be `amount = resources.ChangeCash(amount);` then if I'm not mistaken?

Reply:
I guess. Anything else while Im changing this_

Reply:
I didn't test it ingame yet, the code looks good though.

Reply:
Ok the new code is up.